Layers/Renderers/Building/BuildingRenderer.gd

```diff
@@ -1,84 +1,126 @@
+	# "supermarket": 5
+	preload("res://Resources/Textures/Buildings/PlainWallResources/House.tres"),
+	# "retail_restaurant": 6
+	preload("res://Resources/Textures/Buildings/PlainWallResources/House.tres"),
+	# "historic": 7
+	preload("res://Resources/Textures/Buildings/PlainWallResources/House.tres"),
+	# "religious": 8
+	preload("res://Resources/Textures/Buildings/PlainWallResources/House.tres"),
+	# "greenhouse": 9
+	preload("res://Resources/Textures/Buildings/PlainWallResources/House.tres"),
+	# "concrete": 10
+	preload("res://Resources/Textures/Buildings/PlainWallResources/House.tres"),
+]
+
 var floor_height = 2.5 # Height of one building floor for calculating the number of floors from the height
-var fallback_height = 10
+var fallback_height = 20
+var fallback_num_floors = 2
 var cellar_height = floor_height # For preventing partially floating buildings checked uneven surfaces
+var plinth_height_factor = 0.025
+
+enum flag {
+	basement = 0b1,
+	ground = 0b10,
+	mid = 0b100,
+	top = 0b1000
+}
 
 @onready var height_attribute = layer_composition.render_info.height_attribute_name
 
 
 func _ready():
+	_create_and_set_texture_arrays()
 	super._ready()
 
 
-func load_feature_instance(feature):
-	var polygon = feature.get_outer_vertices()
-	var holes = feature.get_holes()
-
-	var building = building_base_scene.instantiate()
-
-	# Load the components based checked the building attributes
-	var height = util.str_to_var_or_default(
-		feature.get_attribute(height_attribute), fallback_height)
-
-	var num_floors = max(1, height / floor_height)
-
-	# Add a cellar
-	var cellar = plain_walls_scene.instantiate()
-	cellar.set_color(Color.LIGHT_GRAY)
-	building.add_child(cellar)
-
-	# FIXME: Find a way not to have a half window texture here
-#		# In order to get a more accurate height, we add a building base if there's a remainder of
-#		# half a floor height; this also adds some variation and is probably realistic in many cases
-#		if fmod(height, floor_height) > floor_height / 2.0:
-#			var base_floor = plain_walls_scene.instantiate()
-#			base_floor.height = floor_height / 2.0
-#			base_floor.set_window_shading(false)
-#			base_floor.set_color(Color.GRAY)
-#			building.add_child(base_floor)
-
-	# Random facade texture
-	var random_gen = RandomNumberGenerator.new()
-	random_gen.seed = hash(polygon)
-
-	var wall_color = Color.WHITE_SMOKE
-	var random = random_gen.randi_range(0, 10)
-
-	if random >= 0 and random <= 5:
-		wall_color = Color.LIGHT_YELLOW
-	elif random > 5 and random <= 8:
-		wall_color = Color.WHITE_SMOKE
-	elif random == 9:
-		wall_color = Color.DARK_SEA_GREEN
-	elif random == 10:
-		wall_color = Color.LIGHT_BLUE
-
-	# Add the floors
-	for i in range(num_floors):
-		var walls = plain_walls_scene.instantiate()
-		building.add_child(walls)
+# To increase performance, create an array of textures which the same shader can
+# read from
+# TODO: make utility function for this? (Might be useful in other places too)
```

Comment:
I'll probably need this for trees, will generalize it when doing that

Reply:
Did you implement this already? If so, we could probably merge those things to reduce the code base.